### ISSUE TITLE
configure script does not work with bash<4 (i.e., MacOS)

### ISSUE THREAD

**Issue by abouteiller** [ICLDisco/parsec#196]
**[Original report](https://bitbucket.org/icldistcomp/parsec/issue/196) by me.**

----------------------------------------

TL;DR
-----
MaCOS ships by default with a very ancient bash and the configure script uses features that are not compatible with it. 

Description
-----------

The configure script uses *associative bash arrays*, which have been supported for like 15 years. Unfortunately the bash that ships with MacOS is ancient and does not have them. So far I have not encountered another system where this is a compatibility problem, but that may happen. 

Proposed fix
------------

Replace associative arrays with numbered arrays. This will make the code ugly, but it will make it compatible with all variants of bash under the sun.
